api/models/channelSettings.js

```diff
@@ -104,3 +107,58 @@ export const resetChannelJoinToken = (id: string) => {
     .run()
     .then(async () => await getChannelById(id));
 };
+
+type UpdateInput = {
+  channelId: string,
+  slackChannelId: ?string,
+  eventType: 'THREAD_CREATED',
```

Comment:
Why not just make this `eventType: 'threadCreated'`? Then you don't have to do that awkward switch further below to match `threadCreated` to `THREAD_CREATED`, that seems pretty unnecessary.